conf/cassandra.yaml

```diff
@@ -1575,29 +1575,24 @@ drop_compact_storage_enabled: false
 #    - 127.0.0.1
 #    - 127.0.0.0/31
 
-# Enables tracking warnings/aborts across all replicas for reporting back to client.
+# Enables read thresholds (warn/fail) across all replicas for reporting back to the client.
 # See: CASSANDRA-16850
-# track_warnings:
-# Scheduled to enable in 4.2
-#     enabled: true
-# When track_warnings.enabled: true, this tracks the materialized size of a query on the
-# coordinator. If coordinator_large_read.warn_threshold_kb is greater than 0, this will emit a warning
+# read_thresholds_enabled: false # scheduled to be set true in 4.2
+# When read_thresholds_enabled: true, this tracks the materialized size of a query on the
+# coordinator. If coordinator_read_size_warn_threshold is defined, this will emit a warning
 # to clients with details on what query triggered this as well as the size of the result set; if
-# coordinator_large_read.abort_threshold_kb is greater than 0, this will abort the query after it
+# coordinator_read_size_fail_threshold is defined, this will fail the query after it
 # has exceeded this threshold, returning a read error to the user.
-#     coordinator_read_size:
-#         warn_threshold_kb: 0
-#         abort_threshold_kb: 0
-# When track_warnings.enabled: true, this tracks the size of the local read (as defined by
-# heap size), and will warn/abort based off these thresholds; 0 disables these checks.
-#     local_read_size:
-#         warn_threshold_kb: 0
-#         abort_threshold_kb: 0
-# When track_warnings.enabled: true, this tracks the expected memory size of the RowIndexEntry
-# and will warn/abort based off these thresholds; 0 disables these checks.
-#     row_index_size:
-#         warn_threshold_kb: 0
-#         abort_threshold_kb: 0
+# coordinator_read_size_warn_threshold:
+# coordinator_read_size_fail_threshold:
+# When read_thresholds_enabled: true, this tracks the size of the local read (as defined by
+# heap size), and will warn/fail based off these thresholds; undefined disables these checks.
+# local_read_size_warn_threshold:
+# local_read_size_fail_threshold:
+# When read_thresholds_enabled: true, this tracks the expected memory size of the RowIndexEntry
+# and will warn/fail based off these thresholds; undefined disables these checks
+# row_index_size_warn_threshold:
+# row_index_size_fail_threshold:
```

Comment:
Since we are renaming to read thresholds, shouldn't these be named `row_index_read_size_warn_threshold`/`row_index_read_size_fail_threshold`? I think we only check the index size when reading, but there is no limit when writing.

Reply:
```
case ROW_INDEX_SIZE_FAIL:
                    reason = RequestFailureReason.READ_SIZE;
```

our error code says its READ_SIZE, so makes sense, will do

Reply:
pushed change... there is prob a regression somewhere =D